library/dash/src/main/java/com/google/android/exoplayer2/source/dash/manifest/DashManifestParser.java

```diff
@@ -799,9 +817,33 @@ protected SegmentTemplate buildSegmentTemplate(
       long duration,
       List<SegmentTimelineElement> timeline,
       UrlTemplate initializationTemplate,
-      UrlTemplate mediaTemplate) {
+      UrlTemplate mediaTemplate,ArrayList<Descriptor> adaptationSetSupplementalProperties,
+      ArrayList<Descriptor> representationSupplementalProperties ) {
+
+    if (representationSupplementalProperties != null) {
```

Comment:
All other non-trivial code is in `parseSegmentTemplate`, with `buildSegmentTemplate` just for the trivial building part. So this probably belongs in `parseSegmentTemplate`, and `buildSegmentTemplate` should probably be changed only to have an additional `long endNumber` argument.